web/src/activity_ui.js

```diff
@@ -22,6 +21,9 @@ import * as util from "./util";
 export let user_cursor;
 export let user_filter;
 
+// Function initialized from `ui_init` to avoid importing narrow.js and causing circular imports.
+let narrow_by_email;
```

Comment:
@timabbott can you confirm if this commit works in the way you'd like? I'll integrate it into the main commit once you've given feedback.

Reply:
Yep, this is fine.